variables.tf

```diff
@@ -0,0 +1,82 @@
+# ------------------------------------------------------------------------------
+# Variables
+# ------------------------------------------------------------------------------
+variable "name_prefix" {
+  description = "A prefix used for naming resources."
+}
+
+variable "vpc_id" {
+  description = "The ID of the VPC that this container will run in, needed for the Target Group"
+}
+
+variable "private_subnet_ids" {
+  description = "A list of private subnets inside the VPC"
+  type        = "list"
+}
+
+variable "cluster_id" {
+  description = "AWS ECS Cluster id"
+}
+
+variable "task_definition_image" {
+  description = "Image for the task definition (repo:tag or repo@digest)."
+}
+
+variable "lb_arn" {
+  description = "Arn for the LB for which the service should be attach to."
+}
+
+variable "task_container_desired_count" {
+  description = "The number of instances of the task definitions to place and keep running."
+  default     = "1"
+}
+
+variable "task_container_assign_public_ip" {
+  description = "Assigned public IP to the container."
+  default     = "false"
+}
+
+variable "task_container_port" {
+  description = "Port that the container exposes."
+}
+
+variable "task_container_protocol" {
+  description = "Optional: Protocol that the container exposes."
+  default     = "HTTP"
+}
+
+variable "task_definition_cpu" {
+  description = "Optional: Amount of CPU to reserve for the task."
+  default     = "256"
+}
+
+variable "task_definition_memory" {
+  description = "Optional: Amount of memory to reserve for the task."
```

Comment:
Since we are publishing to the terraform registry these days, we have changed a couple of things that would be nice to fix in the descriptions here:
- No need to specify "Optional: " because the registry already lists all variables with default values under a  header called "optional".
- I/we have started using the descriptions from the official documentation as much as possible. So it's easier for users to recognise which variable in a module maps to which variable in a resource, and also helps us identify which variables are the same across modules.

This applies for all the variables in this file. And you should be able to copy-paste most of the descriptions from here:
https://github.com/telia-oss/terraform-aws-ecs/blob/master/modules/service/variables.tf